portfolio/src/main/webapp/leave-comment/script.js

```diff
@@ -0,0 +1,37 @@
+function getComments() {
+  console.log("Downloading comments.");
+  const responsePromise = fetch("/get-comments");
+  responsePromise.then(handleResponse);
+}
+
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+  const JSONPromise = textPromise.then(JSON.parse);
+  // When the response is converted to text, pass the result into the
+  // addCommentsToDom() function.
+  JSONPromise.then(addCommentsToDom);
+}
+
+function addCommentToHTML(commentJSON) {
+  $('#comment-list')
+  .append($('<div class="comment-author"></div>').text(commentJSON.author))
+  .append($('<div class="comment-content"></div>').text(commentJSON.content));
+
+  $('#comment-list > .comment-author').each(function(){
```

Comment:
I find this very confusing: as far as I understand it, `addCommentToHTML` adds `div.comment-author` to `#comment-list` and then `div.comment-content`  to the `div.comment-author`. The `each` iterator then iterates over the `div.comment-autor` elements, selects the underlying `div.comment-content` selects both `div` elements and wraps them in `div.comment`?

if this is the case, we could use [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) and do it in one step.

Reply:
Ok, template strings might impose a security issue. Can we combine the `div.comment` part with the previous one at least? DOM manipulation is very costly, esp. adding nodes between two nodes.

Reply:
"DOM manipulation is very costly" - what kind of cost are you referring to?

Reply:
performance-wise on client side. It's not a problem so far, but if you load 10s of comments, that's an issue.

Reply:
@leon-kastler-google At first I didn't correctly understood your description. The effect I want to achieve looks like this:
`<div class="comment">
    <div class="comment-author"> text </div>
    <div class="comment-content"> text </div>
</div>`
The main problem for me was "wrapping" two divs with big <div class="comment">, this loop at the end is just my workaround with jQuery. Another solution is just to add all nodes by hand (I showed it to you during standup), but I'll do more research if it's possible to make it clean and nice without jQuery.

Reply:
Oh, now I see what you mean. How about creating the whole DOM structure first and then appending it whole to the document in one step?

Reply:
In jQuery you can build a nested elements structure like this:
```
$('body').append(
  $('<div class="comment">')
    .append($('<p class="comment-author">').text('mvv'))
    .append($('<p class="comment-content">').text('hey!'))
)
```

Reply:
Please take a look at my newest change, I think it's a bit better now. And it got shorter :) Thank you very much your advice, I understand JS and jQuery better now.